src/LogFileReader.cpp

```diff
@@ -40,19 +46,42 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+
+
     // TODO implement this first
+    QStringList column = line.split(BATDATA_DELIMITER);
+    if(column.length() != 3)
+        {
+           return false;
+        }
+     QString timeString = column.at(0);
+
+     batteryData.time = QTime::fromString(timeString, STRING_TO_QTIME);
+
+     bool currentOkay;
+     bool voltageOkay;
+
+     batteryData.voltage = column.at(1).toDouble(&voltageOkay);
+     batteryData.current = column.at(2).toDouble(&currentOkay);
 
+     if(!currentOkay || !voltageOkay || !batteryData.time.isValid())
+     {
+         return false;
+     }
+     else
+     {
+         return true;
+     }
     // This is here to the compiler happy. Otherwise the compile
     // will have an error warning about an unused variable. Remove this
     // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+    // Q_UNUSED(line);
+    //Q_UNUSED(batteryData);
     return true;
```

Comment:
You are already returning something previously in an `if else` statement, so this line is no longer necessary.